File: src/Cxx/Utilities/KnownLengthArray.cxx
#include <vtkSmartPointer.h>
#include <vtkFloatArray.h>

int main(int, char *[])
{
  vtkSmartPointer<vtkFloatArray> distances =
    vtkSmartPointer<vtkFloatArray>::New();
  distances->SetName("Distances");
  distances->SetNumberOfComponents(1);
  distances->SetNumberOfValues(5);

  //set values
  for(vtkIdType i = 0; i < distances->GetNumberOfTuples(); i++)
  {
    float f = (float)i + 0.1;
    distances->SetValue(i, f);
  }

  //get values
  for(vtkIdType i = 0; i < distances->GetNumberOfTuples(); i++)
  {
    float f = distances->GetValue(i);
    std::cout << f << std::endl;
  }

  return EXIT_SUCCESS;
}

File: src/Java/Utilities/KnownLengthArray.java
import vtk.vtkNativeLibrary;
import vtk.vtkFloatArray;

public class KnownLengthArray 
{
  //-----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
 //-----------------------------------------------------------------
 public static void main(String s[]) 
 {	
    vtkFloatArray Distances = new vtkFloatArray();
    Distances.SetName("Distances");
    Distances.SetNumberOfComponents(1);
    Distances.SetNumberOfValues(5);
    
    //set values
    for(int i = 0; i < Distances.GetNumberOfTuples(); i++)
    {
      double f = i + 0.1;
      Distances.SetValue(i, f);
    }
    
    //get values
    for(int i = 0; i < Distances.GetNumberOfTuples(); i++)
    {
      double f = Distances.GetValue(i);
      System.out.println(f);
    }		
  
  }
}
